server/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java

```diff
@@ -71,9 +71,6 @@ public class ContextIndexSearcher extends IndexSearcher implements Releasable {
      */
     private static final int CHECK_CANCELLED_SCORER_INTERVAL = 1 << 11;
 
-    // don't create slices with less than 50k docs
-    private static final int MINIMUM_DOCS_PER_SLICE = 50_000;
-
     // make sure each slice has at least 10% of the documents as a way to limit memory usage and
     // to keep the error margin of terms aggregation low
     private static final double MINIMUM_DOCS_PERCENT_PER_SLICE = 0.1;
```

Comment:
Just to double check, this doesn't need to be configurable in tests?

Reply:
I don't think is necessary. It might prevent in some cases to have a thread per segment but that should be ok.

Reply:
More over, it might make results more predictable?

Reply:
Right, I think so too.

Reply:
Is the reasoning that we are always going to make sure that each slice has at least 10% of the documents, otherwise we'd run into unpredictable precision errors for some aggs? What value do you foresee setting this setting to in practice? For a very low value like 1, doesn't the default min docs percentage per slice always win against the min docs per slice test setting value?

Also some minor thought: if we are always providing 1, does the setting need to be a number? Are there cases where we'd provide a different value?